src/app/admin-panel/admin-panel/subjects/subjects.component.ts

```diff
@@ -1,77 +1,132 @@
-import { Component, OnInit } from '@angular/core';
-import { AdminSubjectsService } from 'src/app/services/admin-subjects.service';
-import {Subject} from 'src/app/models/subjects.model';
+import { Component, OnInit } from "@angular/core";
+import { AdminSubjectsService } from "src/app/services/admin-subjects.service";
+import { Subject } from "src/app/models/subjects.model";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
 
 @Component({
-  selector: 'app-subjects',
-  templateUrl: './subjects.component.html',
-  styleUrls: ['./subjects.component.scss']
+  selector: "app-subjects",
+  templateUrl: "./subjects.component.html",
+  styleUrls: ["./subjects.component.scss"]
 })
 export class SubjectsComponent implements OnInit {
+  subjectForm: FormGroup;
 
-	loading: boolean;
-
-	displayDialog: boolean;
-
-	subject: any;
-
-	selectedSubject: Subject;
-
-	newSubject: boolean;
-
-	subjects: Subject[];
-
-	cols: Array<object>;
-
-	constructor(private _subjectsService: AdminSubjectsService) {}
-
-	ngOnInit() {
-		this.loading = true;
-		this._subjectsService.getSubjectsList().subscribe(data => this.subjects = data);
-		this.cols = [{
-				field: 'subjectName',
-				header: 'Назва'
-			},
-			{
-				field: 'subjectDescription',
-				header: 'Опис'
-			}
-		];
-		this.loading = false;
-	}
-
-	showDialogToAdd() {
-		this.newSubject = true;
-		this.subject = {};
-		this.displayDialog = true;
-	}
-
-	create() {
-		this.displayDialog = false;
-		this._subjectsService.postSubject(this.subject).subscribe(
-			subject => this.subjects.push(subject),
-			err => console.error(err)
-		);
-	}
-
-	save() {
-		this.displayDialog = false;
-		this._subjectsService.putSubject(this.subject).subscribe(
-			subject => {
-				let subjects = [...this.subjects];
-				subjects[this.subjects.indexOf(this.selectedSubject)] = subject;
-				this.subjects = subjects;
-			},
-			err => console.error(err)
-		);
-		this.subject = null;
-	}
-
-	onRowSelect(rowData) {
-		this.newSubject = false;
-        this.subject = {
-            ...rowData
-        };
-		this.displayDialog = true;
-	}
+  submitted = false;
+
+  loading: boolean;
+
+  displayDialog: boolean;
+
+  subject: any;
+
+  selectedSubject: Subject;
+
+  newSubject: boolean;
+
+  subjects: Subject[];
+
+  cols: Array<object>;
+
+  constructor(
+    private _subjectsService: AdminSubjectsService,
+    private formBuilder: FormBuilder
+  ) {}
+
+  ngOnInit() {
+    this.loading = true;
+    this._subjectsService
+      .getSubjectsList()
+      .subscribe(data => (this.subjects = data));
+    this.cols = [
+      {
+        field: "subjectName",
+        header: "Назва"
+      },
+      {
+        field: "subjectDescription",
+        header: "Опис"
+      }
+    ];
+    this.loading = false;
```

Comment:
i believe it must be inside subscription

Reply:
Yeah, you are right. 
I replaced it.
Thx)